src/main/kotlin/org/jitsi/videobridge/util/ArrayTracker.kt

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright @ 2019 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.util
+
+import java.lang.Integer.max
+import org.jitsi.nlj.util.floorMod
+import java.lang.Integer.min
+
+/**
+ * Implements a fixed-sized tracker based on a pre-filled array. The main use-case is the frame projection tracker.
+ *
+ * @author Jonathan Lennox, based on [org.jitsi.nlj.util.ArrayCache] by Boris Grozev
+ */
+open class ArrayTracker<T>(
```

Comment:
What are the differences between `ArrayTracker` and `ArrayCache`? Would it be hard to extract the common logic to avoid duplication?

Reply:
The main difference is that `ArrayTracker` clears out all entries older than `lastIndex - size`, whereas `ArrayCache` doesn't.  (The other differences are all reconcilable, I think.)

Reply:
This could probably be added as a conditional on `ArrayCache`, or made possible for a subclass -- do you think it's worth it?